cms/core/migrations/0006_update_user_groups.py

```diff
@@ -0,0 +1,182 @@
+        content_type=content_type,
+        codename=permission_codename,
+        defaults={"name": f"Can {' '.join(permission_codename.split('_'))}"},
+    )
+    group = Group.objects.get(name=group_name)
+
+    group.permissions.add(permission)
+
+
+def create_user_groups(apps):
+    """Remove the existing Wagtail user groups and create new ones,
+    and give them  the "Access the Wagtail admin" permission.
+    """
+    Group = apps.get_model("auth.Group")
+
+    # Remove all existing Wagtail groups
+    Group.objects.all().delete()
+
+    # Create the Publishing Admin, Publishing Officer and Viewer groups and give them the 'access_admin' permission
+    for group_name in [
+        settings.PUBLISHING_ADMINS_GROUP_NAME,
+        settings.PUBLISHING_OFFICERS_GROUP_NAME,
+        settings.VIEWERS_GROUP_NAME,
+    ]:
+        Group.objects.create(name=group_name)
+        assign_permission_to_group(apps, group_name, "access_admin", "wagtailadmin", "admin")
+
+
+def create_reporting_permissions(apps):
+    """Allow Publishing Admins to view the log entry reports."""
+    # The "Reports" section and everything under it
+    assign_permission_to_group(
+        apps, settings.PUBLISHING_ADMINS_GROUP_NAME, "view_logentry", "wagtailcore", "PageLogEntry"
+    )
+
+
+def create_image_permissions(apps):
+    """Grant Publishing Admins full permissions on the root image collection,
+    and allow Publishing Officers to select images for use on pages.
+    """
+    Group = apps.get_model("auth.Group")
+    Permission = apps.get_model("auth.Permission")
+    Collection = apps.get_model("wagtailcore.Collection")
+    GroupCollectionPermission = apps.get_model("wagtailcore.GroupCollectionPermission")
+
+    root_collection = Collection.objects.get(depth=1)
+
+    # Grant add, change and delete permissions to Publishing Admins
+    for permission in WAGTAIL_PERMISSION_TYPES:
+        GroupCollectionPermission.objects.create(
+            group=Group.objects.get(name=settings.PUBLISHING_ADMINS_GROUP_NAME),
+            collection=root_collection,
+            permission=Permission.objects.get(content_type__app_label="wagtailimages", codename=f"{permission}_image"),
+        )
+
+    # Grant "choose" permission to Publishing Admins and Publishing Officers
+    for group_name in [settings.PUBLISHING_ADMINS_GROUP_NAME, settings.PUBLISHING_OFFICERS_GROUP_NAME]:
+        group = Group.objects.get(name=group_name)
+        GroupCollectionPermission.objects.create(
+            group=group,
+            collection=root_collection,
+            permission=Permission.objects.get(content_type__app_label="wagtailimages", codename="choose_image"),
+        )
+
+
+def create_snippet_permissions(apps):
+    """Allow Publishing Admins to create, edit and delete snippets."""
+    snippet_classes = [
+        # app, model, add publish permission
+        ("core", "GlossaryTerm", False),
+        ("core", "ContactDetails", False),
```

Comment:
this is missing:

 Object | Role | Permissions
--------|------|--------------
 Glossary terms | PO | add
 Glossary terms | PA | add, change
 Team | PO | view
 Team | PA | view

Reply:
Regarding the first two: 
- The matrix doesn't mention the PO being able to add a Term (as in create one), only "Add to Content" and this is handled by the Page permissions
- PA has both `add` and change `change` as they're in the `WAGTAIL_PERMISSION_TYPES` list 

With the latter two, me and Mebin agreed to leave that as the next steps, but since the Teams PR has gone in in the meantime I'm happy to add that here.